### DISCUSSION TITLE
Welcome to simplex-chat Discussions!

### DISCUSSION THREAD

**Discussion by epoberezkin** [simplex-chat/simplex-chat#1430] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by trymeouteh** [comment]
Great app, but are there any plans to rewrite the app in dart for flutter. Flutter allows you to compile apps for Android, iOS, Windows, Mac and Linux all with one codebase. I would highly reccomend it since with a flutter app, you can have one codebase for all devices.

**Reply by epoberezkin** [reply]
Hi there, no plans to rewrite it in Flutter or any other cross-platform framework:
1) having one codebase is not something that has value to the end users - we only do things that benefit the users. This equation might change once many developers start integrating the app’s UI parts into their apps - the developers will be the end users then.
2) iOS and Android are very different platforms in many important aspects, e.g. how notifications are handled - in iOS we use notification service extension and in Android we use background service - these would have to be implemented natively as plugins.
3) UX conventions are also different - while some parts of the app have the same or similar UI some other have different UI.

Overall, for a messaging app the current design seems better (= less development effort than any cross platform framework) - a shared core (in our case it’s written in Haskell) that is used by two different native UIs. We did some early spikes with Flutter, it wasn’t looking too promising, tbh.

**Comment by cthompstone** [comment]
Sorry to hijack this thread, couldn't see how to start a new one...
I trying to use the Simplex CLI with a websocket to Node-Red.
the connection works fine.  Message from phone to a group appears in Node-Red.
i just want a simple way to send a message from Node-Red to the websocket CLI and onto to my phone.
I'd like to send it to a group that's already been created.

Now I can do this directly on the CLI, but when I want to send a message via the WS to CLI, looks like I have to generate a hugely complex JSON object.  Since I'm new to this simplex job, and really only code in Node-Red, python and a bit of JS/HTML , I'm struggling really....

Is there any info you can give me as to the JSON format to send a message to a group.  I looked at the docs on here about the protocol, but it's big and not really sure which bits apply to my needs.

**Reply by epoberezkin** [reply]
there is typescript SDK that does it: https://github.com/simplex-chat/simplex-chat/tree/stable/packages/simplex-chat-client/typescript, it might be a bit out of date.

**Comment by LiberteIV** [comment]
Thank you for this amazing chat app, people like you are what keeps the internet a free place as in freedom.
This project is even better than signal as it doesn't have any identifiers like forced phone number, thank you very much!

Starred!

**Reply by zen0bit** [reply]
[Discover – Simplex Public Address Book](https://github.com/oSoWoSo/SimpleX-Discover)

**Comment by RpiController** [comment]
I too am excited and thankful for all the work being put into this project.
Can someone please explain how SimpleX is different from Session? Seems like they are basically identical.
I think there needs to be more publicity about such differences if this project is to get traction. 
Are there any plans for a youtube channel where features could be explained and highlighted?

**Reply by epoberezkin** [reply]
Replied below

**Comment by epoberezkin** [comment]
1. Session has persistent user identifiers, Simplex does not.
2. Session allows to access the same account from multiple devices, at the cost of compromising security - you can’t track such accesses. SimpleX doesn’t yet, the new desktop beta is available, but it will be a separate profile for now.
3. Session protects your IP from your contacts. SimpleX users who self-host relays can observe senders IP addresses. It’ll be improved soon, until then you must use Tor if that’s important to protect your IP addresses from your contacts.
4. SimpleX uses double ratchet for e2e encryption that provides forward secrecy, post compromise security and repudiation (deniability) - Session doesn’t, as it replaced double ratchet with some proprietary protocol.

So these are two very different products with a set of trade offs.

**Reply by RpiController** [reply]
Good to know! Thanks for the honest and detailed response. The absence of persistent user identifiers seems to be SimpleX's standout feature.
Session uses a long difficult-to-remember "recovery phrase" for restoring/moving an account on different devices. Is that the plan for SimpleX as well, or will there be something more 'user-friendly'?

**Reply by epoberezkin** [reply]
No, as having such a phrase is a security risk - it effectively allows anybody to access your data without you knowing.

**Reply by epoberezkin** [reply]
and also send messages as you.

**Comment by Flexible78** [comment]
Cool app. But when issue win release, beta may be, and next - in beta versions video calling freez and show static Pic ((.

**Comment by http403** [comment]
I'm gauging what's my next IM platform would be and one recommend SimpleX. After I read the whitepaper, I have some question that I would like to be addressed. I haven't tried the app yet so they might seems silly.

1. Does a server communicates with other servers? If not, how does two recipents exchange information at the fist hand. For example, Alice want to chat with Bob. How does Alice know where to find Bob?
2. Assuming two recipients Alice and Bob. Alice is using the default server where Bob is using a self-hosted server via Tor. Can Alice chat with Bob?
3. Since simplexmq or SMP itself is assuming to run on top of a trasport, can the transport be I2P?

And some generic question
1. How do I build the Android app as I don't see clear instructions nor in the [CI pipeline](https://github.com/simplex-chat/simplex-chat/blob/fde3c4f4e0aff5e6fd5ae7facba0bb254e0be785/.github/workflows/build.yml). While I found scripts about APK building. There aren't clear instructions on doing myself. I'm not a Android dev so I would like some clarifications.
3. Why SimpleX choose to host a F-Droid repository? Does SimpleX have anything that's not compatible with [F-Driod's Inclusion Policy](https://f-droid.org/en/docs/Inclusion_Policy/)? Or there are other reasons? I would like to know.

Thanks.

P.S. Why I can't open a new discussion?

**Reply by atlasveldine** [reply]
A1. No, they do not. Invite links contain server addresses to facilitate connections. There's more information on this in the docs, somewhere. 
A2. Yes, as long as Alice is using a Tor proxy, even if she has set it to "When available" rather than "Required."
A3. Hypothetically, alternative networks can work. I'm unfamiliar with I2P, though. However, if you can run a network as a SOCKS proxy, it should be useable, in theory.

B1. This is a common complaint and I'm just as clueless as you.
B2. I'm confused. SimpleX _is_ available on F-Droid's repository. [F-Droid's SimpleX](https://f-droid.org/en/packages/chat.simplex.app/)

**Reply by http403** [reply]
A1. So each party needs to exchange contact out of band in order to start the conversation?

A2. So SimpleX doesn't have a build in Tor clients and must rely on Orbot.

B2. On the README.md, the F-Droid icon is pointed to a [private repository](https://app.simplex.chat) hence the question. Still, thanks for the link to F-Droid official repo.

**Comment by mabonce** [comment]
Is there a way to disable the auto text completion suggestions? I run into the issue of them getting accepted when i hit the space bar but my texting is being inserted prior to the added text.

**Comment by chmac** [comment]
Seems like I need to post here before I can post other discussions?

I'm trying to transfer to a new iPhone and it asks for my database password, but I can’t see how to get that from my old phone. Should I post a new discussion about this question?

**Reply by Lord-KA** [reply]
AFAIU, the database is always encrypted, but the default encryption password is stored on the device and applied automatically. I don't think you can extract it, but you can set a custom password and then export the db.

**Reply by chmac** [reply]
Thanks. I posted #3564 and it seems the key can't be extracted. I will have to uninstall and reinstall the app I think, then export and import the database.

It's odd that I can post issues but not start discussions...

**Comment by ilya-corneli** [comment]
IMPORTANT! Your application has a memory leak, on two modern devices, on the popular PC platform, the Simplex application over time occupies all the RAM, personally, your application kept 46 GB of RAM!

**Reply by Lord-KA** [reply]
Maybe fill in an issue with reproducer?

**Reply by ilya-corneli** [reply]
this happens when the application is blocked and running and not closed, it probably happens anyway.  windows

**Comment by securegh** [comment]
Hello @epoberezkin 

I found a fix to Taskbar icon not showing up with Linux Appimage ![sc1](https://github.com/simplex-chat/simplex-chat/assets/113610808/f3034a64-c677-4774-9f57-2ae043c717af)
It's simple just add `StartupWMClass=chat-simplex-desktop-MainKt` entry to the .desktop  file
Now I can pin simplex to taskbar and it looks nice ![sc2](https://github.com/simplex-chat/simplex-chat/assets/113610808/86fd57cf-c757-4090-8743-3ea5d87b3d76)

Thank you for making this awesome app!

**Comment by wajenzi** [comment]
Hello,  I am looking to integrate with webhooks or similar.

**Comment by carljmoss** [comment]
Are there any tools to help fix database problems?

My computer lost power while I was using SimpleX. Now, when I restart and after entering the password I get: 

SQLite3 returned Error
Can't open while attempting to exec
PRAGMA busy_timeout = 100
PRAGMA foreign_keys = ON
PRAGMA trusted_schema = OFF
PRAGMA secure_delete = ON
PRAGMA auto_vacuum = FULL
unable to open database file

and an invitation to open the database folder. I presume the database has been left locked or has been corrupted.

I have absolutely no knowledge of SQLite, so if anyone has a good idea what to do to recover this, please tell me!

**Reply by wallart1** [reply]
Backups

**Comment by Cyrus2-TheGreat** [comment]
Hey, the APK download isn't working. 

Redmi Note 12 Pro+
Android 14

![Screenshot_2024-08-30-08-23-58-506_com.android.chrome-edit.jpg](https://github.com/user-attachments/assets/ccabfdb1-88b1-4bc0-8038-7dcaf48690c2)

![Screenshot_2024-08-30-08-25-09-554_com.github.android-edit.jpg](https://github.com/user-attachments/assets/d24c8733-214e-4661-b684-a7d06a6cf209)

**Reply by Narasimha-sc** [reply]
works for me

**Reply by Cyrus2-TheGreat** [reply]
Well, I downloaded it on the PC, and I got it

**Comment by iBobik** [comment]
Hi, I like to see serious competition to Signal. I see use-case where no other secure chats focus on: teams and communities (where Slack and Discord are popular)

This could also be helpful for the network effect - In an existing community or team only a few members need to explore Simplex and they will onboard rest of the team.

For them there is clear competitive advantage - no other encrypted chat is useable for teams and communities.

They will need just this:
- Threads
- Channels

Competing with Slack and Discord in the similar feature set is long journey, but some teams require security so much that they can do big compromises. So big compromises that they run companies on a Signal, which is hell without threads. 

Does it make sense to you?

**Comment by Szenni** [comment]
Great app :)
Is it possible to play audio files faster or to integrate this function (2x 3x 4x or dynamic)?

**Comment by SEdu11** [comment]
hello am by name serge ngiriyumva  am a student form Bugema University in three year I would like to contribute on this project us an open source student

**Comment by martin-braun** [comment]
When deploying the server to Linode using https://github.com/simplex-chat/marketplace-apps/tree/app/simplex-chat/apps/linode-marketplace-simplex, how to ensure everything stays up2date?

**Comment by FailSpy** [comment]
Hi there, it seems like I'm unable to make a new discussion. Is there a reason for this policy configuration?

Noticed the show and tell category and would like to post about my SimpleX project and see others, but seems like this Discussions section is rather dead. 

I'm on the [simplex-devs group](https://simplex.chat/contact#/?v=1-4&smp=smp%3A%2F%2FPQUV2eL0t7OStZOoAsPEV2QYWt4-xilbakvGUGOItUo%3D%40smp6.simplex.im%2FvYCRjIflKNMGYlfTkuHe4B40qSlQ0439%23%2F%3Fv%3D1-2%26dh%3DMCowBQYDK2VuAyEAHNdcqNbzXZhyMoSBjT2R0-Eb1EPaLyUg3KZjn-kmM1w%253D%26srv%3Dbylepyau3ty4czmn77q4fglvperknl4bi2eb2fdy2bh4jxtf32kf73yd.onion&data=%7B%22type%22%3A%22group%22%2C%22groupLinkId%22%3A%22PD20tcXjw7IpkkMCfR6HLA%3D%3D%22%7D) mentioned in this project's README.md, but that seems quiet as well. I think opening up and supporting this project's GitHub Discussions section would allow for the curious devs that haven't committed enough to download the app to explore around and be more "lured in" as it were. Developer usage, particularly with things like bots, motivate projects into getting adopted by regular users.

**Comment by Mirtk-Srdja-Plandr** [comment]
Hello I am here because SimpleX-chat Appimage is laggy and I want the problem to be solved.

**Reply by Narasimha-sc** [reply]
> Hello I am here because SimpleX-chat Appimage is laggy and I want the problem to be solved.

I agree it tends to be slow, especially if your computer has worse specifications, it heavily depends on CPU and disk speed, what are your specifications?

**Reply by Mirtk-Srdja-Plandr** [reply]
It is not about speed it is about importing a .zip saved profile. I tried it many times on AppImage, and just waited forever.

**Comment by jehoshua7** [comment]
Two questions please:

1. If I join a (non secret) group, does that mean all people in the group can see my one time link ?
2. I'm about to re-install the OS soon, I usually do full backups, then wipe everything, so I looked in the simplex FAQ's and
	elsewhere, no guides about recovering it seems.

	files are in ~/.config
		~/.local/share/simplex/

	so on a re-install, maybe install simplex, then overwrite files in those paths with the backup ??

**Reply by Narasimha-sc** [reply]
> Two questions please:
> 
>     1. If I join a (non secret) group, does that mean all people in the group can see my one time link ?
> 
>     2. I'm about to re-install the OS soon, I usually do full backups, then wipe everything, so I looked in the simplex FAQ's and
>        elsewhere, no guides about recovering it seems.
>        files are in ~/.config
>        ~/.local/share/simplex/
>        so on a re-install, maybe install simplex, then overwrite files in those paths with the backup ??

1. it does not show your one-time link in groups, it wouldn't make any sense to do so, as one-time link can only be used once, so multiple people seeing it is not helpful, it does not show your simplex address in groups either.

2. please see: https://simplex.chat/docs/guide/chat-profiles.html#move-your-chat-profiles-to-another-device it applies to making backups too, but moving your files from ~/.config and ~/.local/share/simplex also works.

**Reply by jehoshua7** [reply]
Great, thank you. So it does seem the settings are not included in the database or chat profile export/import. However I can manually copy those back/restore after a fresh OS install.

**Comment by kostix** [comment]
I have a question regarding testing of an `smp-server` I've depoyed on my own VPS (I see no option of starting a proper new Q&A discussion, so writing here).

So, I've deployed Tor with a dedicated instance and with the settings recommended [there](https://simplex.chat/docs/server.html#quick-start), deployed the `smp-server`, punched the necessary holes in the firewall, started the server to listed on 5223/TCP (I don't want to make it listen on 443), verified the server answers using `nc`.

I then downloaded the Andrond client 6.4.11 (337) from F-Droid, added my server by its `smp://...` address and tapped Test. The test failed with a cryptic error message:

> Test failed at step Connect. Error: chat.simplex.common.model.AgentErrorType$BROKER@6f24ae0

I tinkered with the network settings in the client a bit:

* Set "Private routing" to "never" (in an attempt to bypass using Tor, or at least minimize it).
* Set "Use web port" to off.

This made no difference.

What's puzzling is that if, right after tapping "Test server" I go to the box and run `ss -t`, I clearly see the client has established a TCP session (exactly like when I do `nc`), which stays active for some time and then goes away - this appears to correlate with the client posting that error message dialog.

The problem is that nothing gets logged when the client apparently attempts its connection test - neither in the server's logs, nor in the Tor's, and I see no options to crank up verbosity in the server's config.

So, the question is: how do I troubleshoot testing server from Android client? What are my options?

Another point which bothers me is: does testing somehow try to verify that "web" facet of the server (which I intended to not care about for now)? In other words, what exactly is being tested besieds the fact a TCP connection gets established?

**Comment by abolition123** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Hi,

Found this app, liked it immediately. Proud to sponsor it!

I am planning on making it via my own private server for family and friends.

Looking forward to maybe have some guidance on adding default personal servers with fallback to own servers aswell and making it work smoothly. Any help is greatly appreciated.

I believe .hs files habe to be edited and use nix aswell ... 

Anyone interested in helping is welcome.... 🙏❤️✌️

**Comment by ghost** [comment]
Hey. My name's Stan. I started a few days ago trying to get familiar with the SimpleX Chat messenger and installing my own server. However I have some troubles, some questions and some feedback I would like to share. Unfortunately I do not seem to be able to create a new discussion in the "Discussions" section, despite seeing some discussions already existing there from other users. I do not want to spam this topic here with my problems and questions. How can I create a new discussion there?

**Reply by Narasimha-sc** [reply]
hello, for discussions it is better to use the app itself and join the public groups, for suggestions, bug reports, you can create an issue here too.

**Comment by RudraNarapogu** [comment]
Hello Sir this is Rudra
iam Student and im here for a request 
claude giving free max for those who has 5000 starts in their github repo ..can you add me as a contributor for one of your repo's ,,i will inform you if i get verified for free max plan ..its a request from a student ..can you help me to get into it

**Reply by TON618OFF** [reply]
Does Claude Code give you access to some kind of premium subscription plan if you are a contributor to a repository with 5,000 stars? <br>
What are the limits on this plan?

**Reply by RudraNarapogu** [reply]
No Restrictions You just need to be a contributor to the Repository
